deps/wazuh_testing/wazuh_testing/tools/sockets.py

```diff
@@ -50,3 +51,24 @@ def send_active_response_message(active_response_command):
     sock.connect(ACTIVE_RESPONSE_SOCKET_PATH)
     sock.send(f"{active_response_command}".encode())
     sock.close()
+
+
+def wait_for_tcp_port(port, host='localhost', timeout=10):
+    """Wait until a port starts accepting TCP connections.
+    Args:
+        port (int): Port number.
+        host (str): Host address on which the port should be listening. Default 'localhost'
+        timeout (float): In seconds. How long to wait before raising errors.
+    Raises:
+        TimeoutError: The port isn't accepting connection after time specified in `timeout`.
+    """
+    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    start_time = time.time()
+    while time.time() - start_time < timeout:
+        try:
+            sock.connect((host, port))
+            return
+        except ConnectionRefusedError:
+            time.sleep(1)
```

Comment:
```suggestion
        try:
            sock.connect((host, port))
            return
        except ConnectionRefusedError:
            time.sleep(1)
        finally:
            sock.close()
```

Don't forget to close the socket if it's no longer necessary.

Reply:
This causes an error sometimes when the finally is executed and the socket isn't actually connected:

```
OSError: [Errno 9] Bad file descriptor
```

But I have added the close statement at the end of the try block.